apps/front/pages/trip.vue

```diff
@@ -0,0 +1,326 @@
+      <AppTextWriter
+        id="loadingSentences"
+        :duration="0.5"
+        :sentences="loadingSentences"
+        :interval="1000"
+      />
+    </h2>
+  </div>
+</template>
+
+<script setup lang="ts">
+import vue3StarRatings from 'vue3-star-ratings';
+import { z } from 'zod';
+
+const PlaceSchema = z.object({
+  imgRef: z.string(),
+  name: z.string().nullable(),
+  rating: z.number().nullable().optional(),
+  formatted_address: z.string().nullable().optional(),
+  international_phone_number: z.string().nullable().optional(),
+  website: z.string().nullable().optional(),
+  editorial_summary: z
+    .object({
+      overview: z.string().nullable().optional(),
+    })
+    .nullable()
+    .optional(),
+});
+
+const StepSchema = z.object({
+  start_location: z.object({
+    lat: z.number(),
+    lng: z.number(),
+  }),
+  end_location: z.object({
+    lat: z.number(),
+    lng: z.number(),
+  }),
+  polyline: z.string(),
+});
+
+const RouteSchema = z.object({
+  name: z.string(),
+  lat: z.number(),
+  long: z.number(),
+  steps: z.array(StepSchema),
+});
+
+const DataSchema = z.object({
+  places: z.array(PlaceSchema),
+  routes: z.array(RouteSchema),
+});
+
+const nmbrResults = ref<number>(null);
+const loading = ref(true);
+const waypointLoading = ref(false);
+const collasped = ref(false);
+const drawerOpen = ref(false);
+const selectedItem = ref(null);
+const places = ref<Array<z.infer<typeof PlaceSchema>>>([]);
+const routes = ref<z.infer<typeof RouteSchema>>({});
+const route = useRoute();
+const { t, tm } = useI18n();
+const runtimeConfig = useRuntimeConfig();
+const apiKey = `${runtimeConfig.public.GOOGLE_API_KEY}`;
+
+const loadingSentences = tm('trip.loadingSentences');
+
+const items = [
+  {
+    label: t('trip.types.all'),
+  },
+  {
+    label: t('trip.types.hotel'),
+  },
+  {
+    label: t('trip.types.activity'),
+  },
+  {
+    label: t('trip.types.bar'),
+  },
+  {
+    label: t('trip.types.restaurant'),
+  },
+];
+
+function showDrawer(item) {
+  selectedItem.value = item;
+  drawerOpen.value = true;
+}
+
+async function addWaypoint(item) {
+  waypointLoading.value = true;
+  console.log(waypointLoading.value);
+  const waypoint = route.query.waypoints
+    ? `${route.query.waypoints}|${item.formatted_address}`
+    : item.formatted_address;
+  await navigateTo({
+    query: {
+      departure: route.query.departure,
+      arrival: route.query.arrival,
+      price: route.query.price,
+      d_start: route.query.d_start,
+      d_end: route.query.d_end,
+      waypoints: waypoint,
+    },
+  });
+  fetchData();
+}
+
+function openClose() {
+  collasped.value = !collasped.value;
+}
+
+async function fetchData() {
+  const query = {
+    origin: route.query.departure,
+    destination: route.query.arrival,
+    ...(route.query.waypoints ? { waypoints: route.query.waypoints } : {}),
+  };
+  const { data: fetchedData, error } = await useFetch('/api/call', {
+    query,
+    immediate: true,
+    watch: false,
+  });
```

Comment:
useFetch should be use at the root of the setup script, please look at InputMenuAutocomplete component